core/templates/pages/exploration-editor-page/services/lost-changes-modal.controller.spec.ts

```diff
@@ -0,0 +1,83 @@
+
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for LostChangesModalController.
+ */
+
+describe('Lost Changes Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var $log = null;
+  var logSpy = null;
+  var LocalStorageService = null;
+  var ChangesInHumanReadableFormService = null;
+  var explorationId = '0';
+  var lostChanges = [{
+    cmd: 'add_state',
+    state_name: 'State name',
+  }];
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector, $controller) {
+    $log = $injector.get('$log');
+    LocalStorageService = $injector.get('LocalStorageService');
+    ChangesInHumanReadableFormService = $injector.get(
+      'ChangesInHumanReadableFormService');
+
+    logSpy = spyOn($log, 'error').and.callThrough();
+    // Remove spyOn when #8729 be merged, because
+    // ChangesInHumanReadableFormService was refactored in this PR.
+    // Ref: https://github.com/oppia/oppia/pull/8729
+    spyOn(ChangesInHumanReadableFormService, 'makeHumanReadable').and
+      .returnValue({
+        outerHTML: 'mocking'
+      });
+    var $rootScope = $injector.get('$rootScope');
+
+    $uibModalInstance = jasmine.createSpyObj(
+      '$uibModalInstance', ['close', 'dismiss']);
+
+    $scope = $rootScope.$new();
+    $controller(
+      'LostChangesModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        explorationId: explorationId,
+        lostChanges: lostChanges
+      });
+  }));
+
+  it('should evaluates lostChangesHtml when controller is initialized',
+    function() {
+      // Remove expect when #8729 be merged.
+      expect($scope.lostChangesHtml).toBe('mocking');
+      // Uncomment this piece of code below when #8729 be merged.
+      // expect($scope.lostChangesHtml).toBe(
```

Comment:
Typically we do not submit commented out code. Can you remove this?

Reply:
Sure, I'll just wait to #8729 to be merged, because it will fail without the updates made from the PR.

Reply:
Done.